decompose/src/androidMain/kotlin/com/arkivanov/decompose/AssertMainThread.kt

```diff
@@ -0,0 +1,29 @@
+package com.arkivanov.decompose
+
+import android.os.Looper
+
+private val monitor = Any()
+private var mainThreadIdHolder: MainThreadIdHolder? = null
+
+internal actual fun assertMainThread() {
+    val holder =
+        synchronized(monitor) {
```

Comment:
Maybe it's better to use smth like this:

```kotlin
object MainThreadIdHolder {
        val id: Long by lazy {
            getMainThreadId()
        }
}
```

`lazy` implements double-checked locking under the hood by default

Reply:
Hey! Thanks for looking at the code! There are separate implementations for Android and pure JVM.

On Android it [uses](https://github.com/arkivanov/Decompose/pull/259/files#diff-8b28f67ee31542dc4e3af41e14a16104ca5ce66ad9800b6b27ca09ab62ecac24) `Looper` to check if the current thread is the main thread. If the check fails once because `Looper` is not available (e.g. in unit tests), it will not make more attempts and will just by-pass the check.

On JVM it's [tricky](https://github.com/arkivanov/Decompose/pull/259/files#diff-e8d01a5cda1286fb8f273f8baa1cbe8142aa3be2b90d58de867533d9cd5ef81b) - the way you check for the main thread depends on what GUI toolkit is used (e.g. Swing, JavaFX, etc.). So Decompose will actually check for "same" thread - it will remember the first thread id, and compare all other threads with that "original" thread id.

Does it sound good?

Reply:
I got it, you're right. I'm sorry, I was neglected and missed that there is separate implementation for JVM. Thank you for pointing this. Yes, it sounds great now! Btw what do you think about using double-checked locking? Cause synchronized section on each `assertMainThread` still bothering me

Reply:
Ah yeah, forgot about it. Currently, a Double-Checked Locking looks like a premature optimisation. The check shouldn't be used under load, so the performance should be fine I guess. On the other hand, the current implementation makes it more simple and readable.

Reply:
Got you, thank you for explanation! And what do you think about approach with `lazy` delegate, I suggested? In my opinion it is a little bit more easy to understand, because it doesn't use any low-level abstraction such us `synchronized` and it definitely thread-safe by design. 

I hope am not being too annoying :)

Reply:
Great idea with `lazy`, I have updated the PR. Thanks!